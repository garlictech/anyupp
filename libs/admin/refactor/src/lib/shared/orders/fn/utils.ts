import { CrudSdk } from '@bgap/crud-gql/api';

export interface OrderHandlerDeps {
  crudSdk: CrudSdk;
  timestamp: () => number;
}
