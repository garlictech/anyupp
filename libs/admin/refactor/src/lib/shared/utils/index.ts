import { ImageCompressorService } from './services/image-compressor/image-compressor.service';
import { QrGeneratorService } from './services/qr-generator/qr-generator.service';
import { ToasterService } from './services/toaster/toaster.service';

export * from './const';
export * from './enums';
export * from './fn';
export { ImageCompressorService, ToasterService, QrGeneratorService };
