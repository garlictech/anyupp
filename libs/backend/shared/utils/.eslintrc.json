{
  "extends": [
    "../../../../.eslintrc.json"
  ],
  "ignorePatterns": [
    "!**/*"
  ],
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "plugins": [
        "@typescript-eslint"
      ],
      "parserOptions": {
        "project": [
          "libs/backend/shared/utils/tsconfig.*?.json"
        ]
      }
    }
  ]
}
