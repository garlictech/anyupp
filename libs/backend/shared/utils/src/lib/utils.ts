import * as sst from '@serverless-stack/resources';

export const getFQParamNameFromDetails = (
  stage: string,
  appName: string,
  paramName: string,
) => `/${stage}-${appName}/generated/${paramName}`;

export const getFQParamName = (app: sst.App, paramName: string) =>
  getFQParamNameFromDetails(app.stage, app.name, paramName);

export const commonBackendStack = 'common-backend-anyupp';
export const anyuppVpcName = 'AnyuppVpc';
export const anyuppFargateClusterName = 'anyupp-fargate-cluster';
export const anyuppVpcSecurityGroupParamName =
  '/common/AnyuppVpcSecurityGroupName';
export const anyuppVpcIdParamName = '/common/AnyuppVpcId';

export const getCommonVpcNameDev = () =>
  `dev-${commonBackendStack}/${anyuppVpcName}`;

export const getCommonVpcNameProd = () =>
  `prod-${commonBackendStack}/${anyuppVpcName}`;
