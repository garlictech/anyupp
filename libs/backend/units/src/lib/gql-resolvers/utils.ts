import { CrudSdk } from '@bgap/crud-gql/api';
import { DynamoDB } from 'aws-sdk';

export interface UnitsResolverDeps {
  crudSdk: CrudSdk;
  hashGenerator: (arg: string) => string;
  uuidGenerator: () => string;
  tableName: string;
  docClient: DynamoDB.DocumentClient;
}
