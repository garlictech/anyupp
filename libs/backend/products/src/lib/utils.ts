import { CrudSdk } from '@bgap/crud-gql/api';
import { getAllPaginatedData } from '@bgap/gql-sdk';
import { filterNullishGraphqlListWithDefault } from '@bgap/shared/utils';
import { DocumentClient } from 'aws-sdk/clients/dynamodb';
import {
  GroupProduct,
  SearchGroupProductsQueryVariables,
  SearchUnitProductsQueryVariables,
  UnitProduct,
} from '@bgap/domain';
import { Observable } from 'rxjs';

export interface ProductResolverDeps {
  crudSdk: CrudSdk;
  unitProductTableName: string;
  chainProductTableName: string;
  groupProductTableName: string;
  docClient: DocumentClient;
}

export const getUnitIdsFromUnitProductList = (unitProducts: UnitProduct[]) =>
  unitProducts.map(unitProduct => unitProduct.unitId);

export const listGroupProductsForChainProductParent =
  (crudSdk: CrudSdk) =>
  (chainProductId: string): Observable<Array<GroupProduct>> => {
    const input: SearchGroupProductsQueryVariables = {
      filter: { parentId: { eq: chainProductId } },
    };

    return getAllPaginatedData(crudSdk.SearchGroupProducts, {
      query: input,
      options: { fetchPolicy: 'no-cache' },
    }).pipe(filterNullishGraphqlListWithDefault<GroupProduct>([]));
  };

export const listUnitProductsForGroupProductParents =
  (crudSdk: CrudSdk) =>
  (groupProductIds: string[]): Observable<Array<UnitProduct>> => {
    const input: SearchUnitProductsQueryVariables = {
      filter: { or: groupProductIds.map(x => ({ parentId: { eq: x } })) },
    };

    return getAllPaginatedData(crudSdk.SearchUnitProducts, {
      query: input,
      options: { fetchPolicy: 'no-cache' },
    }).pipe(filterNullishGraphqlListWithDefault<UnitProduct>([]));
  };
