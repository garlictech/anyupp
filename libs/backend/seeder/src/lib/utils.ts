import { seededIdPrefix } from '@bgap/shared/fixtures';
import { Role } from '@bgap/domain';

const generateUnitId = (idx: number) => `${seededIdPrefix}unit_${idx}_id`;
const generateProductCategoryId = (unitIdx: number, idx: number) =>
  `${seededIdPrefix}product_category_c${unitIdx}_${idx}_id`;
const generateUnitProductId = (unitIdx: number, idx: number) =>
  `${seededIdPrefix}unit_product_c${unitIdx}_${idx}_id`;
const generateVariantId = (
  unitIdx: number,
  productId: number,
  idx: number,
  type: string,
) =>
  `${seededIdPrefix}${type}_product_variant_c${unitIdx}_p${productId}_${idx}_id`;
const generateOrderId = (idx: number) => `${seededIdPrefix}order_${idx}_id`;
const generateUserId = (idx: number) => `${seededIdPrefix}user_${idx}_id`;
const generateRoleContextId = (idx: number, role: Role) =>
  `${seededIdPrefix}role_context_${idx}_${role}_id`;
const generateAdminRoleContextId = (
  idx: number,
  role: Role,
  username: string,
) => `${seededIdPrefix}admin_role_context_${idx}_${role}_${username}_id`;

export const seedUtils = {
  generateUnitId,
  generateProductCategoryId,
  generateUnitProductId,
  generateVariantId,
  generateOrderId,
  generateUserId,
  generateRoleContextId,
  generateAdminRoleContextId,
};
