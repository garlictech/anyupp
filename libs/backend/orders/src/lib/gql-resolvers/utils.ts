import { pipe } from 'fp-ts/lib/function';
import * as CrudApi from '@bgap/crud-gql/api';
import { Observable } from 'rxjs';
import * as R from 'ramda';
import { shareReplay, switchMap } from 'rxjs/operators';
import { throwIfEmptyValue } from '@bgap/shared/utils';
import { AxiosStatic } from 'axios';

export interface OrderResolverDeps {
  crudSdk: CrudApi.CrudSdk;
  userId: string;
  unitTableName: string;
  currentTime: () => Date;
  random: () => number;
  axiosInstance: AxiosStatic;
}

const getUnitProductHelper = R.memoizeWith(
  (_sdk: CrudApi.CrudSdk, id: string) => id,
  (sdk: CrudApi.CrudSdk, id: string) =>
    pipe(
      sdk.GetUnitProduct({ id }),
      throwIfEmptyValue(`UnitProduct cannot be found: ${id}`),
      shareReplay(1),
    ),
);

export const getUnitProduct =
  (sdk: CrudApi.CrudSdk) =>
  (id: string): Observable<CrudApi.UnitProduct> =>
    getUnitProductHelper(sdk, id);

const getGroupProductHelper = R.memoizeWith(
  (_sdk: CrudApi.CrudSdk, id: string) => id,
  (sdk: CrudApi.CrudSdk, id: string) =>
    pipe(
      sdk.GetGroupProduct({ id }),
      throwIfEmptyValue(`GroupProduct cannot be found: ${id}`),
      shareReplay(1),
    ),
);

export const getGroupProduct =
  (sdk: CrudApi.CrudSdk) =>
  (id: string): Observable<CrudApi.GroupProduct> =>
    getGroupProductHelper(sdk, id);

const getChainProductHelper = R.memoizeWith(
  (_sdk: CrudApi.CrudSdk, id: string) => id,
  (sdk: CrudApi.CrudSdk, id: string) =>
    pipe(
      sdk.GetChainProduct({ id }),
      throwIfEmptyValue(`ChainProduct cannot be found: ${id}`),
      shareReplay(1),
    ),
);

export const getChainProduct =
  (sdk: CrudApi.CrudSdk) =>
  (id: string): Observable<CrudApi.ChainProduct> =>
    getChainProductHelper(sdk, id);

export const getGroupProductOfUnitProduct =
  (sdk: CrudApi.CrudSdk) =>
  (unitProductId: string): Observable<CrudApi.GroupProduct> =>
    pipe(
      getUnitProduct(sdk)(unitProductId),
      switchMap(unitProduct => getGroupProduct(sdk)(unitProduct.parentId)),
    );

const getGeneratedProductHelper = R.memoizeWith(
  (_sdk: CrudApi.CrudSdk, productId: string) => productId,
  (sdk: CrudApi.CrudSdk, productId: string) =>
    pipe(
      sdk.GetGeneratedProduct({ id: productId }),
      throwIfEmptyValue(`GeneratedProduct cannot be found: ${productId}`),
      shareReplay(1),
    ),
);

export const getGeneratedProduct =
  (sdk: CrudApi.CrudSdk) =>
  (productId: string): Observable<CrudApi.GeneratedProduct> =>
    getGeneratedProductHelper(sdk, productId);
