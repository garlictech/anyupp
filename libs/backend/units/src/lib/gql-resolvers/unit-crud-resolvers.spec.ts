/* eslint-disable @typescript-eslint/no-explicit-any */
import {
  createUnitResolver,
  hashPasswords,
  updateUnitRKeeperDataResolver,
} from './unit-crud-resolvers';
import { UnitsResolverDeps } from './utils';
import { tap } from 'rxjs/operators';
import { of } from 'rxjs';
import { maskDate, simpleFixtures } from '@bgap/shared/fixtures';

const getDeps = (): UnitsResolverDeps => ({
  crudSdk: undefined as any,
  hashGenerator: param => `${param} HASHED`,
  uuidGenerator: () => 'UUID',
  tableName: 'TABLENAME',
  docClient: {
    put: jest.fn().mockReturnValue({
      promise: () => Promise.resolve({ Attributes: 'DYNAMODB PUT RESULT' }),
    }),
    update: jest.fn().mockReturnValue({
      promise: () => Promise.resolve({ Attributes: 'UPDATE RESULT' }),
    }),
  } as any,
});

test('hashPasswords', () => {
  expect(
    hashPasswords(param => `${param} HASHED`)(simpleFixtures.getRKeeperUnit()),
  ).toMatchSnapshot('VALID PASSWORDS');
});

test('createUnitResolver', async () => {
  expect(
    await createUnitResolver(getDeps())({
      input: simpleFixtures.getUnit(),
    }).toPromise(),
  ).toMatchSnapshot({
    createdAt: expect.any(String),
    updatedAt: expect.any(String),
  });
});

const rkeeperCases = [
  {
    label: 'update everything',
    fixture: {
      unitId: 'UNITID',
      rkeeperUsername: 'NEW RKEEPER USERNAME',
      rkeeperPassword: 'NEW RKEEPER PASSWORD',
      anyuppPassword: 'NEW ANYUPP PASSWORD',
    },
  },

  {
    label: 'anyupp password only',
    fixture: {
      unitId: 'UNITID',
      anyuppPassword: 'NEW ANYUPP PASSWORD',
    },
  },
  {
    label: 'rkeeper password only',
    fixture: {
      unitId: 'UNITID',
      rkeeperPassword: 'NEW RKEEPER PASSWORD',
    },
  },
];

test.each(rkeeperCases)(
  'updateUnitRKeeperData: $label',
  ({ fixture }, done: any) => {
    const deps = getDeps();
    deps.crudSdk = {
      GetUnit: jest.fn().mockReturnValue(of(simpleFixtures.getRKeeperUnit())),
    } as any;

    updateUnitRKeeperDataResolver(deps)({
      input: fixture,
    })
      .pipe(
        tap(result => {
          expect(result).toMatchSnapshot('result');
          expect((deps.crudSdk.GetUnit as any).mock.calls).toMatchSnapshot(
            'GetUnit',
          );
          expect(
            maskDate((deps.docClient.update as any).mock.calls),
          ).toMatchSnapshot('update');
        }),
      )
      .subscribe(() => done());
  },
);
