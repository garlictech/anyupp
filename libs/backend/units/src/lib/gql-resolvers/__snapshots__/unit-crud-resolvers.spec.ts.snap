// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createUnitResolver 1`] = `
Object {
  "address": Object {
    "address": "Ág u. 1.",
    "city": "Budapest",
    "country": "Magyarország",
    "location": Object {
      "lat": 47,
      "lng": 19,
    },
    "postalCode": "1021",
    "title": "HQ",
  },
  "chainId": "test_chain_1_id",
  "createdAt": Any<String>,
  "description": Object {
    "en": "Test unit #test_unit_1_id description",
    "hu": "Teszt unit #test_unit_1_id leírás",
  },
  "externalId": "restaurantid",
  "groupId": "test_group_1_id",
  "id": "rkeeper-unit",
  "isAcceptingOrders": true,
  "isActive": true,
  "lanes": Array [
    Object {
      "color": "#e72222",
      "id": "lane_01",
      "name": "bár",
    },
    Object {
      "color": "#e123ef",
      "id": "lane_02",
      "name": "konyha",
    },
  ],
  "name": "Késdobáló S",
  "open": Object {
    "from": "1970-01-01",
    "to": "2970-01-01",
  },
  "packagingTaxPercentage": 27,
  "paymentModes": Array [
    Object {
      "method": "cash",
      "type": "cash",
    },
    Object {
      "method": "card",
      "type": "card",
    },
    Object {
      "method": "inapp",
      "type": "stripe",
    },
  ],
  "pos": Object {
    "rkeeper": Object {
      "anyuppPassword": "ANYUPP_PASSWORD HASHED",
      "anyuppUsername": "ANYUPP_USERNAME",
      "endpointUri": "https://testendpoint.ucs.hu/wp-json/vendor/v1",
      "rkeeperPassword": "3943200386403faebf5f80191f5c1c",
      "rkeeperUsername": "781_55_69_939",
    },
    "type": "rkeeper",
  },
  "ratingPolicy": Object {
    "ratings": Array [
      Object {
        "value": 1,
      },
    ],
  },
  "serviceFeePolicy": Object {
    "percentage": 10,
    "taxPercentage": 10,
    "type": "applicable",
  },
  "supportedOrderModes": Array [
    "pickup",
    "instant",
  ],
  "supportedServingModes": Array [
    "inPlace",
    "takeAway",
  ],
  "tipPolicy": Object {
    "percents": Array [
      2,
    ],
  },
  "updatedAt": Any<String>,
}
`;

exports[`hashPasswords: VALID PASSWORDS 1`] = `
Object {
  "address": Object {
    "address": "Ág u. 1.",
    "city": "Budapest",
    "country": "Magyarország",
    "location": Object {
      "lat": 47,
      "lng": 19,
    },
    "postalCode": "1021",
    "title": "HQ",
  },
  "chainId": "test_chain_1_id",
  "description": Object {
    "en": "Test unit #test_unit_1_id description",
    "hu": "Teszt unit #test_unit_1_id leírás",
  },
  "externalId": "restaurantid",
  "groupId": "test_group_1_id",
  "id": "rkeeper-unit",
  "isAcceptingOrders": true,
  "isActive": true,
  "lanes": Array [
    Object {
      "color": "#e72222",
      "id": "lane_01",
      "name": "bár",
    },
    Object {
      "color": "#e123ef",
      "id": "lane_02",
      "name": "konyha",
    },
  ],
  "name": "Késdobáló S",
  "open": Object {
    "from": "1970-01-01",
    "to": "2970-01-01",
  },
  "packagingTaxPercentage": 27,
  "paymentModes": Array [
    Object {
      "method": "cash",
      "type": "cash",
    },
    Object {
      "method": "card",
      "type": "card",
    },
    Object {
      "method": "inapp",
      "type": "stripe",
    },
  ],
  "pos": Object {
    "rkeeper": Object {
      "anyuppPassword": "ANYUPP_PASSWORD HASHED",
      "anyuppUsername": "ANYUPP_USERNAME",
      "endpointUri": "https://testendpoint.ucs.hu/wp-json/vendor/v1",
      "rkeeperPassword": "3943200386403faebf5f80191f5c1c",
      "rkeeperUsername": "781_55_69_939",
    },
    "type": "rkeeper",
  },
  "ratingPolicy": Object {
    "ratings": Array [
      Object {
        "value": 1,
      },
    ],
  },
  "serviceFeePolicy": Object {
    "percentage": 10,
    "taxPercentage": 10,
    "type": "applicable",
  },
  "supportedOrderModes": Array [
    "pickup",
    "instant",
  ],
  "supportedServingModes": Array [
    "inPlace",
    "takeAway",
  ],
  "tipPolicy": Object {
    "percents": Array [
      2,
    ],
  },
}
`;
